src/core/server/elasticsearch/integration_tests/auditor.test.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  createRootWithCorePlugins,
+  createTestServers,
+  TestElasticsearchUtils,
+  TestUtils,
+} from '../../../../test_utils/kbn_server';
+import { httpServerMock } from '../../http/http_server.mocks';
+
+describe('Elasticsearch', () => {
+  let servers: TestUtils;
+  let esServer: TestElasticsearchUtils;
+  let root: ReturnType<typeof createRootWithCorePlugins>;
+
+  beforeAll(async () => {
+    servers = createTestServers({
+      adjustTimeout: (t) => jest.setTimeout(t),
+    });
+    esServer = await servers.startES();
+  });
+
+  beforeEach(async () => {
+    root = createRootWithCorePlugins();
+  });
+
+  afterEach(async () => {
+    await root.shutdown();
+  });
+
+  afterAll(async () => {
+    await esServer.stop();
+  });
+
+  describe('Auditor', () => {
+    it('logs elasticsearch requests done on behalf on the current & internal user', async () => {
```

Comment:
Should be moved to functional tests when the new ES client exposed to the plugins.